RecoTauTag/HLTProducers/src/L1PFJetsMatching.cc

```diff
@@ -0,0 +1,133 @@
+                
+                if ((myJet1.p4()+myJet2.p4()).M()>mjj){
+                    
+                    mjj =(myJet1.p4()+myJet2.p4()).M();
+                    i1 = i;
+                    i2 = j;
+                }
+            }
+        
+        const PFJet &  myJet1 = (PFMatchedJets)[i1];
+        const PFJet &  myJet2 = (PFMatchedJets)[i2];
+        
+        if ((myJet1.p4().Pt() >= pt1) && (myJet2.p4().Pt() > pt2) && (mjj > Mjj))
+        {
+            
+            Output.first.push_back(myJet1);
+            Output.first.push_back(myJet2);
+            
+        }
+        
+        if ((myJet1.p4().Pt() < pt1) && (myJet1.p4().Pt() > pt2) && (myJet2.p4().Pt() > pt2) && (mjj > Mjj))
+        {
+            
+            const PFJet &  myJetTest = (PFMatchedJets)[0];
+            if (myJetTest.p4().Pt()>pt1){
+                Output.second.push_back(myJet1);
+                Output.second.push_back(myJet2);
+                Output.second.push_back(myJetTest);
+                
+            }
+        }
+        
+    }
+    
+    return Output;
+    
+}
+
+L1PFJetsMatching::L1PFJetsMatching(const edm::ParameterSet& iConfig):
+  jetSrc_    ( consumes<PFJetCollection>                     (iConfig.getParameter<InputTag>("JetSrc"      ) ) ),
+  jetTrigger_( consumes<trigger::TriggerFilterObjectWithRefs>(iConfig.getParameter<InputTag>("L1JetTrigger") ) ),
+  pt1_Min   ( iConfig.getParameter<double>("pt1_Min")),
+  pt2_Min   ( iConfig.getParameter<double>("pt2_Min")),
+  mjj_Min   ( iConfig.getParameter<double>("mjj_Min"))
+{  
+  produces<PFJetCollection>("TwoJets");
+  produces<PFJetCollection>("ThreeJets");
+}
+L1PFJetsMatching::~L1PFJetsMatching(){ }
+
+void L1PFJetsMatching::produce(edm::StreamID iSId, edm::Event& iEvent, const edm::EventSetup& iES) const
+{
+    
+  unique_ptr<PFJetCollection> pfMatchedJets(new PFJetCollection);
+    std::pair<PFJetCollection,PFJetCollection> output;
+    
+  double deltaR    = 1.0;
+  double matchingR = 0.5;
+  
+  // Getting HLT jets to be matched
+  edm::Handle<PFJetCollection > pfJets;
+  iEvent.getByToken( jetSrc_, pfJets );
+        
+  edm::Handle<trigger::TriggerFilterObjectWithRefs> l1TriggeredJets;
+  iEvent.getByToken(jetTrigger_,l1TriggeredJets);
+                
+  //l1t::TauVectorRef jetCandRefVec;
+  l1t::JetVectorRef jetCandRefVec;
+  l1TriggeredJets->getObjects( trigger::TriggerL1Jet,jetCandRefVec);
+
+  math::XYZPoint a(0.,0.,0.);
+        
+ //std::cout<<"PFsize= "<<pfJets->size()<<endl<<" L1size= "<<jetCandRefVec.size()<<std::endl;
+ for(unsigned int iJet = 0; iJet < pfJets->size(); iJet++){
+    for(unsigned int iL1Jet = 0; iL1Jet < jetCandRefVec.size(); iL1Jet++){
+      // Find the relative L2pfJets, to see if it has been reconstructed
+      const PFJet &  myJet = (*pfJets)[iJet];
+    //  if ((iJet<3) && (iL1Jet==0))  std::cout<<myJet.p4().Pt()<<" ";
+      deltaR = ROOT::Math::VectorUtil::DeltaR(myJet.p4().Vect(), (jetCandRefVec[iL1Jet]->p4()).Vect());
```

Comment:
please use deltaR2 from DataFormats/Math/interface/deltaR unless some other functionality is intended